sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -42,7 +47,18 @@ public void grow(int neededSize, UnsafeRow row) {
     final int length = totalSize() + neededSize;
     if (buffer.length < length) {
       // This will not happen frequently, because the buffer is re-used.
-      final byte[] tmp = new byte[length * 2];
+      final byte[] tmp;
+      try {
+        tmp = new byte[length * 2];
+      } catch (NegativeArraySizeException e) {
+        String errorMessage =
+          "NegativeArraySizeException is triggered. The current length is " + buffer.length +
+          ". The new length is " + length + " * 2. totalSize is " + totalSize() +
```

Comment:
Might be helpful to catch the exception here/move current logging to
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/UnsafeRowWriter.java#L196

and log some more stuff and the entire row. Is the whole thing corrupt?